sympy/assumptions/ask.py

```diff
@@ -9,66 +9,1114 @@
+
+        Examples
+        ========
+
+        >>> from sympy import Q, ask, pi
+        >>> ask(Q.odd(0))
+        False
+        >>> ask(Q.odd(2))
+        False
+        >>> ask(Q.odd(3))
+        True
+        >>> ask(Q.odd(pi))
+        False
+
+        """
+        return Predicate('odd')
+
+    @memoize_property(predicate_storage)
+    def prime(self):
+        """
+        Prime number predicate.
+
+        ``ask(Q.prime(x))`` is true iff ``x`` is a natural number greater
+        than 1 that has no positive divisors other than ``1`` and the
+        number itself.
+
+        Examples
+        ========
+
+        >>> from sympy import Q, ask
+        >>> ask(Q.prime(0))
+        False
+        >>> ask(Q.prime(1))
+        False
+        >>> ask(Q.prime(2))
+        True
+        >>> ask(Q.prime(20))
+        False
+
+        """
+        return Predicate('prime')
+
+    @memoize_property(predicate_storage)
+    def composite(self):
+        """
+        Composite number predicate.
+
+        ``ask(Q.composite(x))`` is true iff ``x`` is a positive integer and has
+        at least one positive divisor other than ``1`` and the number itself.
+
+        Examples
+        ========
+
+        >>> from sympy import Q, ask
+        >>> ask(Q.composite(0))
+        False
+        >>> ask(Q.composite(1))
+        False
+        >>> ask(Q.composite(2))
+        False
+        >>> ask(Q.composite(20))
+        True
+
+        """
+        return Predicate('composite')
+
+    @memoize_property(predicate_storage)
+    def commutative(self):
+        """
+        Commutative predicate.
+
+        ``ask(Q.commutative(x))`` is true iff ``x`` commutes with any other
+        object with respect to multiplication operation.
+
+        TODO: Add examples
+
+        """
+        return Predicate('commutative')
+
+    @memoize_property(predicate_storage)
+    def is_true(self):
+        """
+        Generic predicate.
+
+        ``ask(Q.is_true(x))`` is true iff ``x`` is true. This only makes
+        sense if ``x`` is a predicate.
+
+        Examples
+        ========
+
+        >>> from sympy import ask, Q, symbols
+        >>> x = symbols('x')
+        >>> ask(Q.is_true(True))
+        True
+
+        """
+        return Predicate('is_true')
+
+    @memoize_property(predicate_storage)
+    def symmetric(self):
+        """
+        Symmetric matrix predicate.
+
+        ``Q.symmetric(x)`` is true iff the square matrix ``x`` is equal to
+        its transpose. Every square diagonal matrix is a symmetric matrix.
+
+        Examples
+        ========
+
+        >>> from sympy import Q, ask, MatrixSymbol
+        >>> X = MatrixSymbol('X', 2, 2)
+        >>> Y = MatrixSymbol('Y', 2, 3)
+        >>> Z = MatrixSymbol('Z', 2, 2)
+        >>> ask(Q.symmetric(X*Z), Q.symmetric(X) & Q.symmetric(Z))
+        True
+        >>> ask(Q.symmetric(X + Z), Q.symmetric(X) & Q.symmetric(Z))
+        True
+        >>> ask(Q.symmetric(Y))
+        False
```

Comment:
It probably won't be hard to add some simpler handlers to make these facts work with Matrix. When we do we should add some examples to these with actual matrices.